### ISSUE TITLE
Background not loaded in version 0.42

### ISSUE THREAD

**Issue by davc0n** [hyprwm/Hyprland#7319]
### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit 9a09eac79b85c846e3a865a9078a3f8ff65a9259  (props: bump version to 0.42.0).
Date: Wed Aug 7 19:17:10 2024
Tag: v0.42.0, commits: 5069

flags: (if any)


System Information:
System name: Linux
Node name: thinkpad
Release: 6.10.4-arch2-1
Version: #1 SMP PREEMPT_DYNAMIC Sun, 11 Aug 2024 16:19:06 +0000


GPU information: 
03:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Picasso/Raven 2 [Radeon Vega Series / Radeon Vega Mobile Series] [1002:15d8] (rev d2) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

======Config-Start======


Config File: /home/davide/.config/hypr/hyprland.conf: Read Succeeded
monitor=eDP-1,preferred,1920x1080,1.2
monitor=HDMI-A-1,preferred,0x0,1

exec-once = waybar
exec-once = wl-paste --type text --watch cliphist store
exec-once = wl-paste --type image --watch cliphist store

$terminal = kitty --single-instance
$menu = wofi --show drun --define key_expand=space

env = XCURSOR_SIZE,24
env = HYPRCURSOR_SIZE,24

debug {
    disable_logs = false
}

misc {
    force_default_wallpaper = 0
}

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = it
    kb_options = caps:escape

    follow_mouse = 1

    touchpad {
        natural_scroll = no
    }

    sensitivity = 0 # -1.0 to 1.0, 0 means no modification.
}

general {
    layout = dwindle
    resize_on_border = true
    
    gaps_in = 0 
    gaps_out = 0 
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    blur {
        enabled = true
        size = 3
        passes = 1
    }

    drop_shadow = true
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = yes

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    preserve_split = yes # you probably want this
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = on
}

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = suppressevent maximize, class:.* # You'll probably like this. 
windowrulev2 = workspace emptynm, initialTitle:Mozilla Firefox*
windowrulev2 = workspace emptynm, initialClass:dev.zed.Zed

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = SUPER, Q, exec, $terminal
bind = SUPER, C, killactive,
bind = SUPER, W, exec, $menu
bind = SUPER, M, togglesplit,

# Move focus with mainMod + arrow keys
bind = SUPER, H, movefocus, l
bind = SUPER, J, movefocus, d
bind = SUPER, K, movefocus, u
bind = SUPER, L, movefocus, r

# Switch workspaces with mainMod + [0-9]
bind = SUPER, 1, workspace, 1
bind = SUPER, 2, workspace, 2
bind = SUPER, 3, workspace, 3
bind = SUPER, 4, workspace, 4
bind = SUPER, 5, workspace, 5
bind = SUPER, 6, workspace, 6
bind = SUPER, 7, workspace, 7
bind = SUPER, 8, workspace, 8
bind = SUPER, 9, workspace, 9
bind = SUPER, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = SUPER SHIFT, 1, movetoworkspace, 1
bind = SUPER SHIFT, 2, movetoworkspace, 2
bind = SUPER SHIFT, 3, movetoworkspace, 3
bind = SUPER SHIFT, 4, movetoworkspace, 4
bind = SUPER SHIFT, 5, movetoworkspace, 5
bind = SUPER SHIFT, 6, movetoworkspace, 6
bind = SUPER SHIFT, 7, movetoworkspace, 7
bind = SUPER SHIFT, 8, movetoworkspace, 8
bind = SUPER SHIFT, 9, movetoworkspace, 9
bind = SUPER SHIFT, 0, movetoworkspace, 10

# Example special workspace (scratchpad)
bind = SUPER, S, togglespecialworkspace, magic
bind = SUPER SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = SUPER, mouse_down, workspace, e+1
bind = SUPER, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = SUPER, mouse:272, movewindow
bindm = SUPER, mouse:273, resizewindow

# Volume
binde = , XF86AudioRaiseVolume, exec, wpctl set-volume -l 1 @DEFAULT_AUDIO_SINK@ 5%+
binde = , XF86AudioLowerVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%-
bind = , XF86AudioMicMute, exec, wpctl set-mute @DEFAULT_AUDIO_SOURCE@ toggle
bind = , XF86AudioMute, exec, wpctl set-mute @DEFAULT_AUDIO_SINK@ toggle

# Switch to previous workspace
bind = SUPER, TAB, workspace, previous

# Screen brightness
binde = , XF86MonBrightnessUp, exec, brightnessctl s +5%
binde = , XF86MonBrightnessDown, exec, brightnessctl s 5%-

# Move current workspace to monitor
bind = SUPER SHIFT, page_up, movecurrentworkspacetomonitor, l
bind = SUPER SHIFT, page_down, movecurrentworkspacetomonitor, r

# Lock
bind = SUPER, Print, exec, hyprlock

# Clipboard
bind = SUPER, V, exec, cliphist list | wofi --show dmenu | cliphist decode | wl-copy

# Move active window
bind = SUPER SHIFT, h, swapwindow, l
bind = SUPER SHIFT, j, swapwindow, d
bind = SUPER SHIFT, k, swapwindow, u
bind = SUPER SHIFT, l, swapwindow, r

# Resize
binde = ALT, H, resizeactive, -10 0
binde = ALT, J, resizeactive, 0 10
binde = ALT, K, resizeactive, 0 -10
binde = ALT, L, resizeactive, 10 0

======Config-End========
```

</details>

### Description

Updated hyprland to version `0.42`, workspace background is black.
Rollback to `0.41.2` fixes the issue.

### How to reproduce

1. Install hyprland 0.42
2. Restart hyprland

### Crash reports, logs, images, videos

[hyprland.log](https://github.com/user-attachments/files/16603718/hyprland.log)

**Comment by vaxerski** [comment]
the path for the assets changed. You need to install the assets again

**Comment by vaxerski** [comment]
wait, you don't get `Creating a texture for BGTex` logged? Would likely mean it's disabled in your config (disable_hyprland_logo)

**Comment by Steven--** [comment]
I'm also getting black background with:
```
    force_default_wallpaper = 0 # Set to 0 or 1 to disable the anime mascot wallpapers
    disable_hyprland_logo = false # If true disables the random hyprland logo / anime girl background. :(
```

**Comment by davc0n** [comment]
That's very likely an issue with the arch package.

On arch, the assets directory seems to be `/usr/share/hyprland`.
It looks like no custom assets path is configured in arch's PKGBUILD, so as a workaround I tried to copy of the content at `/usr/share/hypr` which should be the default but didn't work.

Opened a PR to include the path in the error message: https://github.com/hyprwm/Hyprland/pull/7336

**Comment by fox091** [comment]
I can confirm `sudo cp -r /usr/share/hyprland /usr/local/share/hypr` fixes the issue on Arch after restarting Hyprland!

I'm guessing the Arch packagers just missed the change to `hypr`, but I'm not really understanding the move from `/usr/share` to `/usr/local/share`.  The Hyprland release notes only show the subdirectory changing, unless `$DATAROOTDIR` is now different.

Should we open a bug on the Arch package?

**Comment by LRitzdorf** [comment]
> Should we open a bug on the Arch package?

I would say yes :)
I'd do it myself, but I lack an Arch GitLab account.

**Comment by fox091** [comment]
I lack one as well.  I'll email them to get an account made and hopefully can file it tomorrow.

**Comment by JL2210** [comment]
I was questioning why they decided to install everything manually myself. It also doesn't install the lowercase hyprland symlink.

I signed up for a gitlab account last week so I can make a bug. Probably will get around to it tomorrow.

**Comment by fox091** [comment]
I created an issue last night for the default backgrounds not appearing: https://gitlab.archlinux.org/archlinux/packaging/packages/hyprland/-/issues/13

**Comment by JL2210** [comment]
https://gitlab.archlinux.org/archlinux/packaging/packages/hyprland/-/merge_requests/3

**Comment by Zenzi0** [comment]
Sorry to barge in but I've got the same symptom only with non-default wallpapers. The background stays black after login.
I'm also on Archlinux (well Arch based Arcolinux) using the latest git from AUR/Chaotic (I tried both). Downgrading to the latest release 0.42 fixes this issue.
My wallpaper manager (Variety) works just fine and I can switch backgrounds and open the current one just fine. So that part is working. The actual background stays black nonetheless.

Is this related and a packaging issue (in AUR this time)?

Maybe I should also mention that I'm on Nvidia (nvidia-open-dkms) and I tried both with and without the nvidia envs set in hyprland.conf (they don't seem to make a difference anymore, so usually I leave them commented out).
